### PR TITLE
Remove bindings from Inlined tree

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/typer/Inliner.scala [hunk 1]
```diff
@@ -508,9 +506,18 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
...
-      // Take care that only argument bindings go into `bindings`, since positions are
-      // different for bindings from arguments and bindings from body.
-      tpd.Inlined(call, finalBindings, finalExpansion)
+      // Widen TermRefs to bindings
+      val finalExpansion2 = finalExpansion.tpe match {
+        case tpe: TermRef if finalBindings.exists(x => x.symbol == tpe.symbol) =>
+          // If the return type of an inline function is a singleton type of a parameter (like in `DottyPredef.the`)
+          // the expansion will be `{ val x = y; (x: x.type) }`. We ascribe the type to itself widening the
+          // local TermRef of the binding `{ val x = y; (x: y.type) }`.
+          tpd.Typed(finalExpansion, TypeTree(ctx.typer.avoidingType(finalExpansion, finalBindings))).withSpan(finalExpansion.span)
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/typer/Inliner.scala]: Here I had to add some extra logic to avoid type refs of the binding.

**Comment by nicolasstucki** [comment]
Will need rebase once #6066 is merged.

**Comment by nicolasstucki** [comment]
Rebased

**Comment by smarter** [comment]
What's the reason for this change ?

**Comment by nicolasstucki** [comment]
Mostly remove unessesary complexity and some code duplication.

**Comment by nicolasstucki** [comment]
Indeed, we are missing some `Inlined(EmptyTree, ...)` nodes around the RHS of the bindings. I will add these `Inlined` and tests to check those positions.

**Comment by nicolasstucki** [comment]
The positions do have the correct source/span, we already set them correctly. I will need to add some extra assertions on the `Inlined` nodes to check the invariant.

**Pull request closed without merging**